test/common/network/lc_trie_speed_test.cc

```diff
@@ -15,6 +20,39 @@ std::unique_ptr<Envoy::Network::LcTrie::LcTrie<std::string>> lc_trie_nested_pref
 
 namespace Envoy {
 
+static void BM_LcTrieConstruct(benchmark::State& state) {
+  std::unique_ptr<Envoy::Network::LcTrie::LcTrie<std::string>> trie;
+  for (auto _ : state) {
+    trie = std::make_unique<Envoy::Network::LcTrie::LcTrie<std::string>>(tag_data);
+  }
+  benchmark::DoNotOptimize(trie);
+}
+
+BENCHMARK(BM_LcTrieConstruct);
+
+static void BM_LcTrieConstructNested(benchmark::State& state) {
+  std::unique_ptr<Envoy::Network::LcTrie::LcTrie<std::string>> trie;
+  for (auto _ : state) {
+    trie = std::make_unique<Envoy::Network::LcTrie::LcTrie<std::string>>(tag_data_nested_prefixes);
+  }
+  benchmark::DoNotOptimize(trie);
+}
+
+BENCHMARK(BM_LcTrieConstructNested);
+
+static void BM_LcTrieConstructMinimal(benchmark::State& state) {
+  std::vector<std::pair<std::string, std::vector<Envoy::Network::Address::CidrRange>>> tags;
+  tags.emplace_back(std::pair<std::string, std::vector<Envoy::Network::Address::CidrRange>>(
```

Comment:
Could you move it to global variable to match others and to avoid recreating this for every run?

Also, could you add `BM_LcTrieLookupMinimal`?